src/DropDownMenu/DropDownMenu.jsx

```diff
@@ -344,56 +258,27 @@ const DropDownMenu = React.createClass({
     const styles = this.getStyles();
 
     let value;
-    let selectedIndex = this._isControlled() ? null : this.state.selectedIndex;
 
-    if (menuItems && typeof selectedIndex === 'number') {
-      warning(menuItems[selectedIndex],
-        `SelectedIndex of ${selectedIndex} does not exist in menuItems.`);
-    }
-
-    if (valueMember && this._isControlled()) {
-      value = this.props.hasOwnProperty('value') ? this.props.value : valueLink.value;
-      if (menuItems && value !== null && value !== undefined) {
-        for (let i = 0; i < menuItems.length; i++) {
-          if (menuItems[i][valueMember] === value) {
-            selectedIndex = i;
-          }
-        }
-      }
+    if (this.props.hasOwnProperty('value')) {
+      value = this.props.value;
```

Comment:
Since we got rid of all the `valueMember` logic, couldn't we just skip this logic and add `value` to the list of props that's that are destructured on line 249?